lib/libmbedtls/core/tee_lmd_provider.c

```diff
@@ -2049,13 +2116,26 @@ void crypto_cipher_free_ctx(void *ctx, uint32_t algo __maybe_unused)
 	 * could never have succeded above.
 	 */
 	assert(!cipher_get_ctx_size(algo, &ctx_size));
-	mbedtls_cipher_free(ctx);
+#if defined(CFG_CRYPTO_XTS) || defined(CFG_CRYPTO_CTS)
+	if (!(algo == TEE_ALG_AES_XTS || algo == TEE_ALG_AES_CTS))
+		mbedtls_cipher_free(ctx);
+#endif
 	free(ctx);
 }
 
 void crypto_cipher_copy_state(void *dst_ctx, void *src_ctx,
 				uint32_t algo __unused)
 {
+	TEE_Result res __maybe_unused;
+	size_t ctx_size __maybe_unused;
+
+#if defined(CFG_CRYPTO_XTS) || defined(CFG_CRYPTO_CTS)
+	if (algo == TEE_ALG_AES_XTS || algo == TEE_ALG_AES_CTS) {
+		res = cipher_get_ctx_size(algo, &ctx_size);
+		assert(!res);
+		memcpy(dst_ctx, src_ctx, ctx_size);
```

Comment:
I think a `return;` is needed after this line.

Reply:
OK, will add it.